perfkitbenchmarker/configs/storage_utility.py

```diff
@@ -0,0 +1,173 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import numpy
+
+
+class SQLProviderError(Exception):
+  pass
+
+
+class IOPSError(Exception):
+  pass
+
+
+class StorageConfig(object):
+  """Given iops and service provider requirements, return necessary parameters.
+
+  Assumes SSD persistent disks.
+
+  Args:
+    iops: Number of iops required.
+    provider: Provider: 'AWS' or 'GCP'.
+  """
+
+  AWS = 'AWS'
+  GCP = 'GCP'
+
+  def __init__(self, iops, provider=GCP):
+    self._storage_size = None
+    self._cpu_count = None
+    self._number_disks = None
+    self._iops = iops
+    self._provider = provider.upper()
+    self.ValidateProvider()
+    self.ValidateIOPS()
+    self.PopulateConfigs()
+
+  def ValidateProvider(self):
+    """Validate provider to be GCP or AWS, throw exception if invalid.
+
+    Raises:
+      SQLProviderError: Incorrect provider type given.
+    """
+    if self._provider != self.AWS and self._provider != self.GCP:
+      raise SQLProviderError('Provider given is not supported by '
+                             'storage_utility.')
+
+  def ValidateIOPS(self):
+    """Validate IOPS to be positive integer, throw exception if invalid.
+
+    Raises:
+      IOPSError: Invalid IOPS parameter given.
+    """
+    if self._iops < 1:
+      raise IOPSError()
+
+  def PopulateConfigs(self):
+    """Poplulate Storage Configurations."""
+    self.SetStorageSize()
+    self.SetCPUCount()
+    self.SetNumberDisks()
+
+  def PrintConfigs(self):
+    """Print out necessary configs."""
+    vm_config = ("For {} IOPS using {}, the following is required:\n\tStorage "
+                 "Size (GB): {}\n\tCPU Count: {}\n\tNumber of "
+                 "Disks: {}").format(self._iops,
+                                     self._provider.upper(),
+                                     self._storage_size,
+                                     self._cpu_count,
+                                     self._number_disks)
+    print vm_config
+
+  def SetStorageSize(self):
+    """Set Storage size.
+
+    GCP: ratings from https://cloud.google.com/compute/docs/disks/. Storage can
+          go as high as 64TB per disk but IOPS maxes out at 30,000 iops/disk
+          or (1000GB).
+    AWS: ratings from
+          http://docs.aws.amazon.com/AWSEC2/latest/
+          UserGuide/EBSVolumeTypes.html#EBSVolumeTypes_gp2. Storage can go as
+          high as 16TiB per disk but IOPS maxes out at 10000 IOPS/volume size.
+          Reference gives volume size in GiB so converted to GB below.
+    """
+    if self._provider == self.GCP:
+      self._storage_size = min(self._iops / 30, 30000 / 30)
```

Comment:
Good catch.

Added int() around division to eliminate fractional.
Utilized math.ceil() and made 30 a floating point number so it will always round up. A little messier but catches those corner cases.